apps/extension/src/routes/popup/approval/transaction/index.tsx

```diff
@@ -65,11 +75,18 @@ export const TransactionApproval = () => {
       </div>
       <div className='grow overflow-auto p-4'>
         {selectedTransactionViewName === TransactionViewTab.SENDER &&
-          !hasAltGasFee(selectedTransactionView) && (
+          (hasTransparentAddress(selectedTransactionView) ||
+            !hasAltGasFee(selectedTransactionView)) && (
             <div className='mb-4 rounded border border-yellow-500 p-2 text-sm text-yellow-500'>
               <span className='block text-center font-bold'>⚠ Privacy Warning:</span>
-              Transaction uses a non-native fee token. To reduce gas costs and protect your privacy,
-              maintain an UM balance for fees.
+              {hasTransparentAddress(selectedTransactionView) ? (
+                <p>This transaction uses transparent addresses which may reduce privacy.</p>
+              ) : (
+                <p>
+                  Transaction uses a non-native fee token. To reduce gas costs and protect your
+                  privacy, maintain an UM balance for fees
+                </p>
+              )}
```

Comment:
Why is this an either/or? Shouldn't we have two separate warnings, since each case can happen independently? If they both occur they can be stacked on top of each other.

Reply:
sure, we can stack these privacy warnings

Reply:
I wasn't sure if we wanted to display both privacy warnings, so `t-addr` had a higher priority.

Reply:
<img width="512" alt="Screenshot 2024-12-20 at 3 12 49 PM" src="https://github.com/user-attachments/assets/2481acc9-ec0d-450e-8e3c-9053b58ce40e" />

Reply:
<img width="512" alt="Screenshot 2024-12-20 at 3 19 24 PM" src="https://github.com/user-attachments/assets/059e1b84-f250-4d83-bda9-3adcceb117d6" />